test/e2e/key_test.go

```diff
@@ -128,4 +137,98 @@ var _ = Describe("When deploying SecretProviderClass CRD with keys", func() {
 		Expect(err).To(BeNil())
 		Expect(key).To(ContainSubstring(config.KeyValue))
 	})
+
+	It("should read RSA-HSM key from pod", func() {
+		// update the secretproviderclass to reference rsa-hsm keys
+		keyVaultObjects := []provider.KeyVaultObject{
+			{
+				ObjectName: "rsahsmkey1",
+				ObjectType: provider.VaultObjectTypeKey,
+			},
+		}
+
+		yamlArray := provider.StringArray{Array: []string{}}
+		for _, object := range keyVaultObjects {
+			obj, err := yaml.Marshal(object)
+			Expect(err).To(BeNil())
+			yamlArray.Array = append(yamlArray.Array, string(obj))
+		}
+
+		objects, err := yaml.Marshal(yamlArray)
+		Expect(err).To(BeNil())
+
+		spc.Spec.Parameters["objects"] = string(objects)
+		spc = secretproviderclass.Update(secretproviderclass.UpdateInput{
+			Updater:             kubeClient,
+			SecretProviderClass: spc,
+		})
+
+		p = pod.Create(pod.CreateInput{
+			Creator:                  kubeClient,
+			Config:                   config,
+			Name:                     "busybox-secrets-store-inline-crd",
+			Namespace:                ns.Name,
+			SecretProviderClassName:  spc.Name,
+			NodePublishSecretRefName: nodePublishSecretRef.Name,
+		})
+
+		pod.WaitFor(pod.WaitForInput{
+			Getter:         kubeClient,
+			KubeconfigPath: kubeconfigPath,
+			Config:         config,
+			PodName:        p.Name,
+			Namespace:      ns.Name,
+		})
+
+		cmd := getPodExecCommand("cat /mnt/secrets-store/rsahsmkey1")
+		_, err = exec.KubectlExec(kubeconfigPath, p.Name, p.Namespace, strings.Split(cmd, " "))
```

Comment:
Should we check contents of the key?

Reply:
@sozercan We can run some basic validation like ensuring the content is not nil and the `BEGIN PUBLIC KEY` and `END PUBLIC KEY` exists but I don't think we need the overhead of validating the key. WDYT?

Reply:
sgtm

Reply:
done!